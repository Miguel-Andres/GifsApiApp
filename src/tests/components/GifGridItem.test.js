import React from "react"
import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"


describe("Pruebas en <GifGridItem/>" ,()=>{

    const title ="hola soy el titulo test" ;
    const url = "htpp://urlimagetest.com"
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test("debe de mostrar el componente correctamente" ,()=>{

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el titulo', () => { 

        const img = wrapper.find("img")
        //console.log(img.props())

        expect(img.props().alt).toBe(title)
        expect(img.prop("src")).toBe(url)

     })


     test('debe de tener animate__fadeIn', () => { 

        const div= wrapper.find("div")
        const className = div.prop("className")

        expect(className.includes("animate__fadeIn")).toBe(true)

       // expect(div.hasClass("animate__fadeIn")).toBe(false) esta la hice con la documentacion salio OK
        

      })


})