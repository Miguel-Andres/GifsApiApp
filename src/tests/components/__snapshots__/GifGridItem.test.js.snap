// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeIn "
  >
    <img
      alt="hola soy el titulo test"
      src="htpp://urlimagetest.com"
    />
    <p>
      hola soy el titulo test
    </p>
  </div>
</Fragment>
`;
